src/main/java/dev/jbang/source/buildsteps/CompileBuildStep.java

```diff
@@ -42,10 +43,24 @@ protected Project compile() throws IOException {
 		optionList.addAll(project.getMainSourceSet().getCompileOptions());
 		String path = project.resolveClassPath().getClassPath();
 		if (!Util.isBlankString(path)) {
-			optionList.addAll(Arrays.asList("-classpath", path));
+			optionList.add("-classpath");
+			optionList.add(path);
 		}
 		optionList.addAll(Arrays.asList("-d", compileDir.toAbsolutePath().toString()));
 
+		// add -sourcepath for all source folders
+		List<String> srcDirs = project	.getMainSourceSet()
+										.getSources()
+										.stream()
+										.map(x -> x.getFile().getParent().toAbsolutePath())
```

Comment:
if I read this right it is adding blindly folders parent - not taking into consideration the package hierarchy - thus this would add the wrong root afaics?

Reply:
Yes, that's what I realized too, it's also why it's still Draft 😅 

But that unfortunately means that I need to rethink the Project handling. I might need to add the idea of source folders to it, which is somewhat unfortunate because it adds complexity which until now wasn't necessary.

Reply:
Could we do it on just one file per unique folder. To not require it run on multiple files again and again.

Reply:
I'm actually not so much worried about performance but more that the extraction of the information might be ambiguous. We're just doing a very simple match on lines that start with "package" so even a block comment that has the word "package" might match and throw things off. That was something that never bothered me before because at most it would cause some minor problems with the `edit` command, but now it would affect the actual running of the code and any mistake would result in fatal errors. So I'm somewhat hesitant suddenly requiring this information.